resources/js/features/game-list/components/GameListItems/GameListItemElement.tsx

```diff
@@ -0,0 +1,176 @@
+
+  const { auth } = usePageProps();
+
+  const { t } = useLaravelReactI18n();
+
+  const { addToWantToPlayGamesList, isPending, removeFromWantToPlayGamesList } =
+    useWantToPlayGamesList();
+
+  /**
+   * Invalidation of infinite queries is _very_ expensive, both for the
+   * user (needing to refetch all loaded pages in the infinite scroll), and
+   * for us (needing to actually query/load all that data quickly).
+   * To avoid this, we'll use optimistic state.
+   */
+  const [isInBacklogOptimistic, setIsInBacklogOptimistic] = useState(isInBacklog ?? false);
+
+  const handleToggleFromBacklogClick = () => {
+    if (!auth?.user && typeof window !== 'undefined') {
+      window.location.href = route('login');
+
+      return;
+    }
+
+    setIsInBacklogOptimistic((prev) => !prev);
+
+    const mutationPromise = isInBacklogOptimistic
+      ? removeFromWantToPlayGamesList(game.id, game.title, {
+          t_successMessage: t('Removed :gameTitle from playlist!', { gameTitle: game.title }),
+        })
+      : addToWantToPlayGamesList(game.id, game.title, {
+          t_successMessage: t('Added :gameTitle to playlist!', { gameTitle: game.title }),
+        });
+
+    mutationPromise.catch(() => {
+      setIsInBacklogOptimistic(isInBacklog ?? false);
+    });
+  };
+
+  return (
+    <BaseDrawer shouldScaleBackground={false} modal={false}>
+      <div className="flex gap-3">
+        <GameAvatar {...game} size={48} hasTooltip={false} showLabel={false} />
+
+        <div className="flex-grow truncate">
+          <div className="flex flex-col gap-1">
+            <a href={route('game.show', { game: game.id })} className="truncate tracking-tight">
+              <GameTitle title={game.title} />
+            </a>
+
+            <div className="flex items-center gap-1">
+              {game.system ? (
+                <SystemChip {...game.system} className="light:bg-neutral-200/70" />
+              ) : null}
+
+              {sortFieldId ? (
+                <ChipOfInterest
+                  game={game}
+                  playerGame={playerGame ?? undefined}
+                  sortFieldId={sortFieldId}
+                />
+              ) : null}
+            </div>
+          </div>
+        </div>
+
+        <div className="-mr-2.5 flex self-center">
+          <button
+            className="p-2 text-neutral-100 light:text-neutral-950"
+            onClick={handleToggleFromBacklogClick}
+            disabled={isPending}
+          >
+            <MdClose
+              className={cn(
+                'h-4 w-4 transition-transform',
+                'disabled:!text-neutral-100 light:disabled:text-neutral-950',
+                isInBacklogOptimistic ? '' : 'rotate-45',
+              )}
+            />
+
+            <span className="sr-only">
+              {isInBacklogOptimistic
+                ? t('Remove from Want To Play Games')
+                : t('Add to Want to Play Games')}
+            </span>
```

Comment:
removing items doesn't dynamically remove the row in mobile mode.
undoing a remove in non-mobile mode no longer restores the row.

Reply:
Good catch, this should now be fixed in latest.

There is one quirk around this that feels worth noting:
* As a user, I open my Want to Play Games List on mobile.
* I click the three dots icon to pull up the game details sheet.
* I click the button to remove the game from my list.
When the user does these steps, the sheet instantly vanishes rather than smoothly closing by scrolling to the bottom of the screen. This is occurring because the sheet is a child element of the row, and by removing the game, the row is no longer mounted in the DOM.

This quirk is probably fine for now. The UX can be further smoothed out later.